sycl/plugins/unified_runtime/ur/adapters/level_zero/kernel.hpp

```diff
@@ -83,8 +83,10 @@ struct ur_kernel_handle_t_ : _ur_object {
   struct ArgumentInfo {
     uint32_t Index;
     size_t Size;
-    // const ur_mem_handle_t_ *Value;
-    ur_mem_handle_t_ *Value;
+    // ArgValue contains a generic pointer argument
+    void *ArgValue;
+    // MemValue contains a memory pointer argument
+    ur_mem_handle_t_ *MemValue;
```

Comment:
So, it is either USM pointer or buffer, one or the other, right?
Then I suggest to use std::variant here, it will also make it more robust to known the exact value type.